libs/spark/src/Unstable_AvatarButton/Unstable_AvatarButton.tsx

```diff
@@ -20,6 +20,10 @@ type _Unstable_AvatarButtonTypeMap<
     >;
   defaultComponent: D;
   classKey: Unstable_AvatarButtonClassKey;
+  /**
+   * The size of the component.
+   */
+  size?: 'large' | 'medium';
```

Comment:
```suggestion
    > & {
    /**
     * The size of the component.
     */
    size?: 'large' | 'medium';
    };
  defaultComponent: D;
  classKey: Unstable_AvatarButtonClassKey;
```
This is a generic utility type, the props have a specific spot they all have to boil down to.

Reply:
Thanks for this, where did you go to learn things like this? Do you have any resources that helped you? Or was it all just from the mui docs?